pkg/splunk/enterprise/indexercluster.go

```diff
@@ -414,6 +425,39 @@ func (mgr *indexerClusterPodManager) getClusterMasterClient() *splclient.SplunkC
 	return mgr.newSplunkClient(fmt.Sprintf("https://%s:8089", fqdnName), "admin", adminPwd)
 }
 
+func getSiteRepFactor(siteRepFactor string) int32 {
+	re := regexp.MustCompile(".*origin:(?P<rf>.*),.*")
+	match := re.FindStringSubmatch(siteRepFactor)
+	siteRF, _ := strconv.Atoi(match[1])
+	return int32(siteRF)
+}
+
+// verifyRFPeers verifies the number of peers specified in the replicas section
+// of IndexerClsuster CR. If it is less than RF, than we se it to RF.
```

Comment:
Under what scenarios will the idc CR not have .spec.replicas updated? According to [this](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#scale-subresource) shouldn't it be automatically updated?

Reply:
Not sure if I understood your question, but in our case when we tried to call `kubectl scale idxc --replicas=<new_replica_count>`, it did not update the spec since we were bailing out before we even called `updateStatefulSetPods`.

Reply:
The `kubectl scale idxc --replicas=<new_replica_count>` command for idxc CR is supposed to update its `spec.replicas` field for the IDXC CR. `updateStatefulSetPods` is responsible for scaling up/down pods based on this field.

Reply:
As per discussion, we are self-correcting in the case of Replicas < SF, RF and we take care of deploying the right number of indexers. If we already met RF/SF, kubectl scale should work.